### PR TITLE
Zeus - Paradrop Cargo

### PR DESCRIPTION
**Pull request by PabstMirror**
Zeus can command an aircraft to paradrop cargo at a location

### CODE REVIEW

File: addons/zeus/functions/fnc_moduleCargoParadrop.sqf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private _group = group _pilot;
+
+    if ((!alive _vehicle) || {!alive _pilot} || {[_pilot] call EFUNC(common,isPlayer)}) exitWith {WARNING_2("vehicle %1 - pilot %2 are invalid",_vehicle,_pilot);};
+
+    [_group] call CBA_fnc_clearWaypoints;
+    private _wp = _group addWaypoint [ASLtoAGL _mousePosASL, 0];
+    _wp setWaypointType "SCRIPTED";
+    _wp setWaypointScript QPATHTOF(functions\DOUBLES(fnc,moduleCargoParadropWaypoint).sqf);
+    TRACE_2("Waypoint Added",_wp,local _group); // Locality is handled by the waypoint
+}, localize LSTRING(moduleCargoParadrop_DisplayName)] call FUNC(getModuleDestination);
```
Comment (mharis001) [inline addons/zeus/functions/fnc_moduleCargoParadrop.sqf]: Add `_angle` param of 45 to keep consistent with other modules, default was changed to 0.

File: addons/zeus/functions/fnc_moduleCargoParadrop.sqf [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    [format ["%1 %2", localize "str_dn_aircraft", localize "str_msg_no_veh_select"]] call FUNC(showMessage);
+};
+if ((!alive _vehicle) || {!alive _pilot}) exitWith {
+    [LSTRING(OnlyAlive)] call FUNC(showMessage);
+};
+if ([_pilot] call EFUNC(common,isPlayer)) exitWith {
+    ["str_a3_cfgvehicles_moduleremotecontrol_f_errorPlayer"] call FUNC(showMessage);
+};
+if ((_vehicle getVariable [QEGVAR(cargo,loaded), []]) isEqualTo []) exitWith {
+    ["No cargo"] call FUNC(showMessage);
```
Comment (mharis001) [inline addons/zeus/functions/fnc_moduleCargoParadrop.sqf]: Localize

File: addons/zeus/config.cpp [hunk 3]
```diff
@@ -19,6 +19,7 @@ class CfgPatches {
             QGVAR(moduleToggleNvg),
             QGVAR(moduleToggleFlashlight),
             QGVAR(moduleSimulation),
+            QGVAR(moduleCargoParadrop),
```
Comment (mharis001) [inline addons/zeus/config.cpp]: Move down to `GVAR(cargo)` like `moduleLoadIntoCargo`. That handles requiredAddons.

File: addons/zeus/functions/fnc_moduleCargoParadrop.sqf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+if ((!local _logic) || {!_activated}) exitWith {};
+
+// Validate the module target
+private _vehicle = attachedTo _logic;
+private _pilot = driver _vehicle;
+TRACE_4("moduleCargoParadrop placed",_logic,typeOf _vehicle,_pilot,typeOf _pilot);
+
+deleteVehicle _logic; // cleanup logic now, we just needed it to get the attached vehicle
+
+if (!((["ace_cargo"] call EFUNC(common,isModLoaded)) && {missionNamespace getVariable [QEGVAR(cargo,enable), false]})) exitWith {
```
Comment (mharis001) [inline addons/zeus/functions/fnc_moduleCargoParadrop.sqf]: `isModLoaded` check not necessary after moved to proper `CfgPatches` class.

**Comment by alganthe** [comment]
Same thoughts as mharis, tested and works well.

**Comment by PabstMirror** [comment]
I was a little surprised that moving the created waypoint in zeus actually works
Looks like they are modifying the param `_wpPos` by reference

File: addons/zeus/functions/fnc_moduleCargoParadropWaypoint.sqf [hunk 5]
```diff
@@ -38,6 +40,7 @@ waitUntil {
         TRACE_2("move",_vehicleGroup,_movePos);
         _vehicle flyInHeight 150;
         _vehicleGroup move _movePos;
+        systemChat str _wpPos;
```
Comment (alganthe) [inline addons/zeus/functions/fnc_moduleCargoParadropWaypoint.sqf]: you forgot about it maybe ?

**Pull request merged**